x-pack/plugins/aiops/public/components/spike_analysis_table/spike_analysis_table_groups.tsx

```diff
@@ -0,0 +1,210 @@
+  changePoints,
+  groupTableItems,
+  dataViewId,
+  loading,
+  onPinnedChangePoint,
+  onSelectedChangePoint,
+  selectedChangePoint,
+}) => {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const [sortField, setSortField] = useState<keyof GroupTableItem>(DEFAULT_SORT_FIELD);
+  const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>(DEFAULT_SORT_DIRECTION);
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<Record<string, JSX.Element>>(
+    {}
+  );
+
+  const toggleDetails = (item: GroupTableItem) => {
+    const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+    if (itemIdToExpandedRowMapValues[item.id]) {
+      delete itemIdToExpandedRowMapValues[item.id];
+    } else {
+      const { group, repeatedValues } = item;
+
+      const expandedTableItems = [];
+      const fullGroup = { ...group, ...repeatedValues };
+
+      for (const fieldName in fullGroup) {
+        if (fullGroup.hasOwnProperty(fieldName)) {
+          const fieldValue = fullGroup[fieldName];
+          expandedTableItems.push({
+            fieldName: `${fieldName}`,
+            fieldValue: `${fullGroup[fieldName]}`,
+            ...(changePoints.find(
+              (changePoint) =>
+                (changePoint.fieldName === fieldName ||
+                  changePoint.fieldName === `${fieldName}.keyword`) &&
+                (changePoint.fieldValue === fieldValue ||
+                  changePoint.fieldValue === `${fieldValue}.keyword`)
+            ) ?? {}),
+          });
+        }
+      }
+
+      itemIdToExpandedRowMapValues[item.id] = (
+        <SpikeAnalysisTableExpandedRow
+          changePoints={expandedTableItems}
+          loading={loading}
+          onPinnedChangePoint={onPinnedChangePoint}
+          onSelectedChangePoint={onSelectedChangePoint}
+          selectedChangePoint={selectedChangePoint}
+          dataViewId={dataViewId}
+        />
+      );
+    }
+    setItemIdToExpandedRowMap(itemIdToExpandedRowMapValues);
+  };
+
+  const columns: Array<EuiBasicTableColumn<GroupTableItem>> = [
+    {
+      align: RIGHT_ALIGNMENT,
+      width: '40px',
+      isExpander: true,
+      name: (
+        <EuiScreenReaderOnly>
+          <span>Expand rows</span>
+        </EuiScreenReaderOnly>
+      ),
+      render: (item: GroupTableItem) => (
+        <EuiButtonIcon
+          onClick={() => toggleDetails(item)}
+          aria-label={itemIdToExpandedRowMap[item.id] ? 'Collapse' : 'Expand'}
+          iconType={itemIdToExpandedRowMap[item.id] ? 'arrowUp' : 'arrowDown'}
+        />
+      ),
+    },
+    {
+      'data-test-subj': 'aiopsSpikeAnalysisTableColumnGroup',
+      field: 'group',
+      name: i18n.translate(
+        'xpack.aiops.correlations.failedTransactions.correlationsTable.groupLabel',
+        { defaultMessage: 'Group' }
+      ),
+      render: (_, { group }) => (
+        <EuiCodeBlock
```

Comment:
Is the intention for the code block to fill all the available width per row? I am not seeing that:

<img width="1409" alt="image" src="https://user-images.githubusercontent.com/7405507/189927455-0387de37-bfb3-4a02-825a-1b5003bba9db.png">

Reply:
JSON has been removed and badges used instead.